src/System.Data.SqlClient/src/System/Data/Locale/Locale.cs

```diff
@@ -32,7 +32,7 @@ static Locale()
             {
                 throw ADP.ArgumentOutOfRange("lcid");
             }
-            return s_cachedEncodings.GetOrAdd(lcid, GetDetailsInternal);
+            return s_cachedEncodings.GetOrAdd(lcid, id => GetDetailsInternal(id));
```

Comment:
A bug is when the compiler behaves counter to the design.  The compiler is behaving as designed here so not a bug.